test/javascript/components/accordion.js

```diff
@@ -1,20 +1,142 @@
+
+      plugin.up($html.find('.accordion-content').eq(0));
+      $html.find('.accordion-content').eq(0).should.have.attr('aria-hidden', 'true');
+      $html.on('up.zf.accordion', function() {
+        $html.find('.accordion-content').eq(0).should.be.hidden;
+        done();
+      });
+    });
+
+    it('toggles attributes of title of the targeted container', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {});
+
+      plugin.up($html.find('.accordion-content').eq(1));
+      $html.find('.accordion-title').eq(1).should.have.attr('aria-expanded', 'false');
+      $html.find('.accordion-title').eq(1).should.have.attr('aria-selected', 'false');
+    });
+
+    it('not closes the open container if allowAllClosed is false', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {allowAllClosed: false});
+
+      plugin.up($html.find('.accordion-content').eq(0));
+      $html.find('.accordion-content').eq(0).should.be.visible;
+      // Element has aria-hidden="false" not set if it has not been actively toggled so far
+      // Therefor check if it has not aria-hidden="true"
+      $html.find('.accordion-content').eq(0).should.not.have.attr('aria-hidden', 'true');
+    });
+  });
+
+  describe('down()', function() {
+    it('opens the targeted container', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {});
+
+      plugin.down($html.find('.accordion-content').eq(1));
+      $html.find('.accordion-content').eq(1).should.be.visible;
+      $html.find('.accordion-content').eq(1).should.have.attr('aria-hidden', 'false');
+    });
+
+    it('toggles attributes of title of the targeted container', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {});
+
+      plugin.down($html.find('.accordion-content').eq(1));
+      $html.find('.accordion-title').eq(1).should.have.attr('aria-expanded', 'true');
+      $html.find('.accordion-title').eq(1).should.have.attr('aria-selected', 'true');
+    });
+
+    it('closes open container if multiExpand is false', function(done) {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {multiExpand: false});
+
+      plugin.down($html.find('.accordion-content').eq(1));
+      $html.find('.accordion-content').eq(0).should.have.attr('aria-hidden', 'true');
+      $html.on('up.zf.accordion', function() {
+        $html.find('.accordion-content').eq(0).should.be.hidden;
+        done();
+      });
+    });
+
+    it('not closes open container if multiExpand is true', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {multiExpand: true});
+
+      plugin.down($html.find('.accordion-content').eq(1));
+      $html.find('.accordion-content').eq(0).should.be.visible;
```

Comment:
Another place where I think we have to wait for the event before we make the assertion...